h2o-core/src/main/java/water/ExternalFrameWriterBackend.java

```diff
@@ -73,16 +84,6 @@ static void handleWriteToChunk(ByteChannel sock, AutoBuffer ab) throws IOExcepti
         }
         // close chunks at the end
         ChunkUtils.closeNewChunks(nchnk);
-
-        // Flag informing sender that all work is done and
-        // chunks are ready to be finalized.
-        //
-        // This also needs to be sent because in the sender we have to
-        // wait for all chunks to be written to DKV; otherwise we get race during finalizing and
-        // it happens that we try to finalize frame with chunks not ready yet
-        AutoBuffer outputAb = new AutoBuffer();
```

Comment:
We no longer need this hacky confirmation as the initialization, writing to chunks and frame finalization happens here. The external code (=sparkling water) just calls the API methods:
```
writer.initFrame()
writer.writeToChunk()
writer.finalizeFrame()
```

Reply:
At the end we need to confirmation to ensure the request is done on the backend side. Otherwise the client will just continue with work & the back-end might have not finished the current task ( init/ write to chunks/finalize)